rancher/rancher-services/internal-dns-service/index.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Internal DNS Service
+layout: rancher-default
+
+---
+
+## Internal DNS Service
+
+Within Rancher, we have an internal DNS service that allows all services within one [environment]({{site.baseurl}}/rancher/configuration/environments) to resolve to any other in the environment. 
+
+All services in the environment are resolvable by `<service_name>` and there is no linking required between the services. For any services that are in a different stack, you'd resolve by `<service_name>.<stack_name>` instead of just `<service_name>`. If you would like to resolve a service by a different name, you could set a link so that the service could be resolvable by the service alias. 
+
+Any container in a service will get configured with DNS in `/etc/resolv.conf`. Rancher maintains three DNS namespaces.
+
+* Rancher - `rancher.internal` 
+* Stack - `stack_name.rancher.internal` 
+* Service - `service_name.stack_name.rancher.internal` 
+
+### Setting Service Alias via linking
+
+In the UI, when [adding a service]({{site.baseurl}}/rancher/rancher-ui/applications/stacks/adding-services/), expand the **Service Links** section, select the service, and provide the alias name.
+
+If you're using `rancher-compose` to [add the service]({{site.baseurl}}/rancher/rancher-compose/), the `docker-compose.yml` would use either the `links` or `external_links` directive.
+
+```yaml
+service1:
+  image: wordpress
+  # If the other service is in the same stack
+  links:
+    # <service_name>:<service_alias>
+    - service2:mysql
+  # If the other service is in a different stack
+  external_links:
+    # <service_name>:<service_alias>
+    - service3:mysql
+```
+
+### Sidekicks and Linking
+
+We currently do not support links/external_links in a sidekick service. 
```

Comment:
May be we should start with that primary and sidekick services are resolvable to each other by default, and only then mention that creating alias via links is not supported.